src/lib/scan/stateMachine.ts

```
@@ -0,0 +1,73 @@
+      case "processing":
+        state = transitionState(state, "processing");
+        break;
+      case "complete":
+        state = transitionState(state, "complete");
+        break;
+      case "failed":
+        state = transitionState(state, "failed");
+        break;
+      default:
+        state = state;
+    }
```

Biome (2.1.2)

[error] state is assigned to itself.

This is where is assigned.

Self assignments have no effect and can be removed.

(lint/correctness/noSelfAssign)